# Viewing the Data

One of the most used method for getting a quick overview of the DataFrame, is the <b>head()</b> method.

The <b>head()</b> method returns the headers and a specified number of rows, starting from the top.

In [1]:
import pandas as pd

df = pd.read_csv('data.csv')

print(df.head(15)) 

    Duration  Pulse  Maxpulse  Calories
0         60    110       130     409.1
1         60    117       145     479.0
2         60    103       135     340.0
3         45    109       175     282.4
4         45    117       148     406.0
5         60    102       127     300.0
6         60    110       136     374.0
7         45    104       134     253.3
8         30    109       133     195.1
9         60     98       124     269.0
10        60    103       147     329.3
11        60    100       120     250.7
12        60    106       128     345.3
13        60    104       132     379.3
14        60     98       123     275.0


<b>Note:</b> if the number of rows is not specified, the <b>head()</b> method will return the top 5 rows.

In [5]:
import pandas as pd

df = pd.read_csv('data.csv')

print(df.head()) 

   Duration  Pulse  Maxpulse  Calories
0        60    110       130     409.1
1        60    117       145     479.0
2        60    103       135     340.0
3        45    109       175     282.4
4        45    117       148     406.0


# tail()
tail() method for viewing the last rows of the DataFrame.

The <b>tail()</b> method returns the headers and a specified number of rows, starting from the bottom.

In [10]:
import pandas as pd

df = pd.read_csv('data.csv')

print(df.tail()) 

     Duration  Pulse  Maxpulse  Calories
164        60    105       140     290.8
165        60    110       145     300.0
166        60    115       145     310.2
167        75    120       150     320.4
168        75    125       150     330.4


In [11]:
import pandas as pd

df = pd.read_csv('data.csv')

print(df.tail(10)) # print last 10 data 

     Duration  Pulse  Maxpulse  Calories
159        30     80       120     240.9
160        30     85       120     250.4
161        45     90       130     260.4
162        45     95       130     270.0
163        45    100       140     280.9
164        60    105       140     290.8
165        60    110       145     300.0
166        60    115       145     310.2
167        75    120       150     320.4
168        75    125       150     330.4


# Info About the Data

The DataFrames object has a method called <b>info()</b>, that gives you more information about the data set.

In [12]:
import pandas as pd

df = pd.read_csv('data.csv')

# info
print(df.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 169 entries, 0 to 168
Data columns (total 4 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   Duration  169 non-null    int64  
 1   Pulse     169 non-null    int64  
 2   Maxpulse  169 non-null    int64  
 3   Calories  164 non-null    float64
dtypes: float64(1), int64(3)
memory usage: 5.4 KB
None


# Null Values

The <b>info()</b> method also tells us how many Non-Null values there are present in each column, and in our data set it seems like there are 164 of 169 Non-Null values in the "Calories" column.

Which means that there are 5 rows with no value at all, in the "Calories" column, for whatever reason.

Empty values, or Null values, can be bad when analyzing data, and you should consider removing rows with empty values. This is a step towards what is called cleaning data, and you will learn more about that in the next chapters.

##### @jabedkhanjb
visit me: www.allmylinks.com/jabedkhanjb